Ports/Android/src/com/codename1/impl/android/AndroidImplementation.java

```diff
@@ -9025,17 +9025,78 @@ public void setCurrentForm(final Form f) {
         if (getActivity() == null) {
             return;
         }
-        if(getCurrentForm() == null){
+        if (getCurrentForm() == null) {
             flushGraphics();
         }
-        if(editInProgress()) {
+        if (editInProgress()) {
             stopEditing(true);
         }
+        resetInsetsForNewForm();// Clean paddings y listeners before Form
         super.setCurrentForm(f);
-        if (isNativeTitle() &&  !(f instanceof Dialog)) {
+        if (isNativeTitle() && !(f instanceof Dialog)) {
             getActivity().runOnUiThread(new SetCurrentFormImpl(getActivity(), f));
         }
     }
+    
+    private void resetInsetsForNewForm() {
+        final String methodName = "resetInsetsForNewForm";
+        if (Build.VERSION.SDK_INT >= 34) {
+            final Activity activity = getActivity();
+            if (activity == null) {
+                Log.e(methodName, "Activity is null");
+                return;
+            }
+            activity.runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Window window = activity.getWindow();
+                        if (window == null) {
+                            Log.e(methodName, "Window is null");
+                            return;
+                        }
+                        View decorView = window.getDecorView();
+                        ViewGroup content = window.findViewById(android.R.id.content);
+                        if (content == null || content.getChildCount() == 0) {
+                            Log.e(methodName, "Content view is empty");
+                            return;
+                        }
+                        View rootView = content.getChildAt(0);
+                        if (rootView == null || rootView.getWindowToken() == null || !rootView.isAttachedToWindow()) {
+                            Log.e(methodName, "RootView is not ready (null or not attached)");
+                            return;
+                        }
+                        // Restore config by Default
+                        window.setDecorFitsSystemWindows(true);

```

Comment:
This code seems very broad. My concern is that it might fail in some scenarios e.g. when different build hints are used.

What do you think @shannah